File: PointFreeLeetCode/PointFree.LeetCode_CSharp/Problem_674.cs
namespace PointFree.LeetCode_CSharp;

// 674. Longest Continuous Increasing Subsequence
// Problem Link (Practice): https://leetcode.com/problems/longest-continuous-increasing-subsequence
public class Problem_674
{
    public int FindLengthOfLCIS(int[] nums) 
    {
        var totalMax = 1;
        var currentMax = 1;
        for (var i = 1; i < nums.Length; i++)
        {
            if (nums[i] > nums[i - 1])
            {
                currentMax++;
                totalMax = Math.Max(totalMax, currentMax);
            }
            else
            {
                currentMax = 1;
            }
        }    
        
        return totalMax;
    }
}

File: PointFreeLeetCode/PointFree.LeetCode/Problem_674.fs
namespace PointFree.LeetCode

open System

// 674. Longest Continuous Increasing Subsequence
// Problem Link (Practice): https://leetcode.com/problems/longest-continuous-increasing-subsequence
module Problem_674 =

    // Tracks length of specified "trend" in a sequence of values based on a comparison function
    let findLengthOfLCIS: int list -> int =
        List.groupByAdjacent (>)
        >> List.map List.length
        >> List.max
       